File: 2387-partition-array-such-that-maximum-difference-is-k/partition-array-such-that-maximum-difference-is-k.ts
function partitionArray(nums: number[], k: number): number {
    nums.sort((a, b) => a -b);

    let res = 1;
    let mini = nums[0];

    for(let i=1;i<nums.length;i++){
        if(nums[i] - mini > k){
            mini = nums[i];
            res++;
        }
    }

    return res;
};

File: 2387-partition-array-such-that-maximum-difference-is-k/partition-array-such-that-maximum-difference-is-k.cpp
class Solution {
public:
    int partitionArray(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());

        int res = 1;
        int mini = nums[0];
       

        for(int i =1;i<nums.size();i++){
            if(nums[i] - mini <= k ){

            }
            else{
                mini = nums[i];
                res++;
            }
        }

        return res;
    }
};